<a href="https://colab.research.google.com/github/fajarrinaldi88/Data-Science-2026/blob/main/Pertemuan2_FajarRinaldi_250401020204.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

* Nama : Fajar Rinaldi
* NIM : 250401020204
* Kelas : IF405

In [8]:
import pandas as pd
import numpy as np

In [9]:
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/refs/heads/master/titanic.csv"
df = pd.read_csv(URL)

In [10]:
print(df.shape, df.columns.tolist())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [11]:
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [12]:
wk1 = df[(df['Sex'] == 'female') & (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

0.9680851063829787 34.61176470588235


In [13]:
hasil = df.groupby('Pclass').agg(
    total=('PassengerId', 'count'),
    selamat=('Survived', 'mean'),
).round(3)
print(hasil)

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


In [14]:
print(df.groupby('Pclass')['Survived'].mean())

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64


## Kesimpulan :

**Apa yang dipelajari:**

Dari analisis data kapal Titanic, kita mempelajari tentang demografi penumpang dan tingkat kelangsungan hidup mereka berdasarkan beberapa faktor seperti kelas penumpang (Pclass), jenis kelamin (Sex), dan usia (Age).

**Temuan Utama:**

Tingkat Kelangsungan Hidup Tinggi untuk Wanita Kelas 1: Wanita di kelas 1 memiliki tingkat kelangsungan hidup yang sangat tinggi, yaitu sekitar 96.8%. Usia rata-rata mereka adalah 34.6 tahun.

**Keterbatasan atau Pertanyaan yang Muncul:**

Data Hilang: Kolom 'Age' memiliki 177 nilai yang hilang, dan 'Cabin' memiliki 687 nilai yang hilang, sementara 'Embarked' memiliki 2 nilai yang hilang. Ini bisa memengaruhi hasil analisis jika tidak ditangani dengan benar. Untuk analisis yang lebih mendalam, kita perlu mempertimbangkan metode imputasi untuk nilai yang hilang, terutama untuk 'Age'.
